src/components/templates/PartyMap/components/Map/PartyMapRoomOverlay.tsx

```diff
@@ -14,104 +17,205 @@ import RoomAttendance from "../RoomAttendance";
+    !venue.roomVisibility || venue.roomVisibility === RoomVisibility.nameCount;
+
+  // TODO: what is a better semantic name for this?
+  const shouldShow2 =
+    shouldShow1 ||
+    (venue.roomVisibility === RoomVisibility.count && hasAttendance);
+
+  const containerStyles = useMemo(
+    () => ({
+      left: `${room.x_percent}%`,
+      top: `${room.y_percent}%`,
+      width: `${room.width_percent}%`,
+      height: `${room.height_percent}%`,
+    }),
+    [room.height_percent, room.width_percent, room.x_percent, room.y_percent]
+  );
 
   return (
     <div
       className={`room position-absolute ${isSelectedRoom && "isUnderneath"}`}
-      style={{
-        left: left + "%",
-        top: top + "%",
-        width: width + "%",
-        height: height + "%",
-      }}
-      key={room.title}
-      onClick={toggleRoom}
-      onMouseEnter={roomHovered}
-      onMouseLeave={roomUnhovered}
+      style={containerStyles}
+      // TODO: is this unique? Do we even need a key here since we're not directly inside a loop?
+      // key={room.title}
+      onClick={onClick1}
+      onMouseEnter={handleRoomHovered}
+      onMouseLeave={handleRoomUnhovered}
     >
-      <div className={`camp-venue ${isSelectedRoom ? "clicked" : ""}`}>
-        <div className="camp-venue-img">
-          <img src={room.image_url} title={room.title} alt={room.title} />
-        </div>
-        {venue.roomVisibility === RoomVisibility.hover &&
-          isRoomHovered &&
-          isRoomHovered.title === room.title && (
-            <div className="camp-venue-text">
-              <div className="camp-venue-maininfo">
-                <div className="party-map-venue-title">{room.title}</div>
-                <RoomAttendance venue={venue} room={room} />
-              </div>
-            </div>
-          )}
-
-        <div className={`camp-venue-text`}>
-          {(!venue.roomVisibility ||
-            venue.roomVisibility === RoomVisibility.nameCount ||
-            (venue.roomVisibility === RoomVisibility.count &&
-              hasAttendance)) && (
-            <div className="camp-venue-maininfo">
-              {(!venue.roomVisibility ||
-                venue.roomVisibility === RoomVisibility.nameCount) && (
-                <div className="party-map-venue-title">{room.title}</div>
-              )}
-              <RoomAttendance venue={venue} room={room} />
-            </div>
-          )}
+      {/* TODO: this is what Camp's version had: <div className={`camp-venue ${isThisRoomClicked ? "clicked" : ""}`}>*/}
+      {/*<div className={`camp-venue ${isSelectedRoom ? "clicked" : ""}`}>*/}
+      {/* TODO: is this needed here anymore? */}
+      {/*<div className={`grid-room-btn ${isSelectedRoom && "isUnderneath"}`}>*/}
+      {/*  <div*/}
+      {/*      className="btn btn-white btn-small btn-block"*/}
+      {/*      onClick={onJoinRoom}*/}
+      {/*  >*/}
+      {/*    {venue.joinButtonText ?? "Join now"}*/}
+      {/*  </div>*/}
+      {/*</div>*/}
```

Comment:
- [ ] [TODO] this is what `Camp`'s version had: `<div className={`camp-venue ${isThisRoomClicked ? "clicked" : ""}`}`
- [ ] [TODO] is this needed here anymore?

Reply:
resolved in https://github.com/sparkletown/sparkle/pull/840/commits/4b6a1669636861e45d8f7344d9ba82a8437b29c8